### PR TITLE
test: num files to read exceed the max_parquet_fanout and all sorted …

### PR DESCRIPTION
**Pull request by NGA-TRAN**
…files are resorted

Related to [17556](https://github.com/influxdata/idpe/issues/17556#issuecomment-1533580348)

While working on [different issue](https://github.com/influxdata/idpe/issues/17556), I found we resort the already-sorted-files of the number of input files to the plan is over 40. Create this test for us to discuss to reduce the resort in compactor and querier

### CODE REVIEW

File: influxdb_iox/tests/query_tests2/cases/in/duplicates_parquet_50_files.sql.expected [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+| logical_plan    | Aggregate: groupBy=[[]], aggr=[[COUNT(Int64(1)), SUM(m.f1)]]    |
+|    |   TableScan: m projection=[f1]    |
+| physical_plan    | AggregateExec: mode=Final, gby=[], aggr=[COUNT(Int64(1)), SUM(m.f1)]    |
+|    |   CoalescePartitionsExec    |
+|    |     AggregateExec: mode=Partial, gby=[], aggr=[COUNT(Int64(1)), SUM(m.f1)]    |
+|    |       RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1    |
+|    |         ProjectionExec: expr=[f1@1 as f1]    |
+|    |           DeduplicateExec: [tag1@2 ASC,tag2@3 ASC,tag3@4 ASC,tag4@5 ASC,time@6 ASC]    |
+|    |             SortPreservingMergeExec: [tag1@2 ASC,tag2@3 ASC,tag3@4 ASC,tag4@5 ASC,time@6 ASC,__chunk_order@0 ASC]    |
+|    |               SortExec: expr=[tag1@2 ASC,tag2@3 ASC,tag3@4 ASC,tag4@5 ASC,time@6 ASC,__chunk_order@0 ASC]    |
```
Comment (NGA-TRAN) [inline influxdb_iox/tests/query_tests2/cases/in/duplicates_parquet_50_files.sql.expected:22]: This SortExec is not necessary but due to large fan-out it is added.

**Comment by NGA-TRAN** [comment]
I see a lot of warnings like this in queriers but not compactor. So we only apply this `optimization` (but not optimized?) to querier

```log
2023-05-03 15:07:03 | level=warn msg="cannot use pre-sorted parquet files, fan-out too wide" n_files=46 max_parquet_fanout=40 target="iox_query::physical_optimizer::sort::parquet_sortness" location="iox_query/src/physical_optimizer/sort/parquet_sortness.rs:76" time=1683140823973578120
```

File: influxdb_iox/tests/query_tests2/cases/in/duplicates_parquet_50_files.sql.expected (hunk 1 above)
Comment (wolffcm) [reply influxdb_iox/tests/query_tests2/cases/in/duplicates_parquet_50_files.sql.expected:22]: There is 50 rows here up until `DeduplicateExec` which would only output one row, correct?

What is in the `__chunk_order` field, I can see that it in the `projection` of the `ParquetExec`, but how is it assigned?

Comment (NGA-TRAN) [reply influxdb_iox/tests/query_tests2/cases/in/duplicates_parquet_50_files.sql.expected:22]: > There is 50 rows here up until DeduplicateExec which would only output one row, correct?

In this test each file has one row, so yes, 50 rows until deduplicate reduce them to 1.

> What is in the __chunk_order field, I can see that it in the projection of the ParquetExec, but how is it assigned?

I know that we always add the chunk ID (which is unique) to make the order deterministic if the value of previous columns the same. So I think this `__chunk_order` for that value. @alamb and @crepererum knows this better.

**Review by wolffcm — APPROVED** [review]
Looks good to me.

**Pull request merged**